x-pack/legacy/plugins/transform/public/app/sections/create_transform/components/step_define/step_define_form.tsx

```diff
@@ -89,6 +92,58 @@ export function getDefaultStepDefineState(
     valid: false,
   };
 }
+
+export function applyTransformConfigToDefineState(
+  state: StepDefineExposedState,
+  transformConfig?: TransformPivotConfig
+): StepDefineExposedState {
+  // apply the transform configuration to wizard DEFINE state
+  if (transformConfig !== undefined) {
+    // transform aggregations config to wizard state
+    state.aggList = Object.keys(transformConfig.pivot.aggregations).reduce((aggList, aggName) => {
+      const aggConfig = transformConfig.pivot.aggregations[aggName] as Dictionary<any>;
+      const agg = Object.keys(aggConfig)[0];
+      aggList[aggName] = {
+        agg: agg as PIVOT_SUPPORTED_AGGS,
+        aggName,
+        dropDownName: aggName,
+        ...aggConfig[agg],
+      } as PivotAggsConfig;
+      return aggList;
+    }, {} as PivotAggsConfigDict);
+
+    // transform group by config to wizard state
+    state.groupByList = Object.keys(transformConfig.pivot.group_by).reduce(
+      (groupByList, groupByName) => {
+        const groupByConfig = transformConfig.pivot.group_by[groupByName] as Dictionary<any>;
+        const groupBy = Object.keys(groupByConfig)[0];
+        groupByList[groupByName] = {
+          agg: groupBy as PIVOT_SUPPORTED_GROUP_BY_AGGS,
+          aggName: groupByName,
+          dropDownName: groupByName,
+          ...groupByConfig[groupBy],
+        } as PivotGroupByConfig;
+        return groupByList;
+      },
+      {} as PivotGroupByConfigDict
+    );
+
+    // only apply the query from the transform config to wizard state if it's not the default query
+    const query = transformConfig.source.query;
+    if (query !== undefined && !isEqual(query, { match_all: {} })) {
```

Comment:
I guess it would be better to compare with `defaultQuery`, already seen it somewhere. Or create a function `isDefaultQuery`

Reply:
`isDefaultQuery` is already used but refers to a search using `*`. I created `matchAllQuery/isMatchAllQuery` and consolidated its use across the plugin in c5ae56a.